freeipa/src/test/java/com/sequenceiq/freeipa/service/instance/InstanceMetaDataServiceTest.java

```diff
@@ -156,9 +157,40 @@ public void testSaveInstanceAndGetUpdatedStack() {
                 .filter(im -> INSTANCE_PRIVATE_ID_3 == im.getPrivateId())
                 .findFirst().get();
         assertEquals("ipa3.dom", instanceMetaData.getDiscoveryFQDN());
+        verify(multiAzCalculatorService).filterSubnetByLeastUsedAz(instanceGroup, subnetAzMap);
         verify(multiAzCalculatorService).updateSubnetIdForSingleInstanceIfEligible(subnetAzMap, subentUsage, instanceMetaData, instanceGroup);
     }
 
+    @Test
+    public void testSaveInstanceAndGetUpdatedStackWhenNoAvailabilityZoneDataAvailable() {
+        FreeIpa freeIpa = new FreeIpa();
+        freeIpa.setHostname("ipa");
+        freeIpa.setDomain("dom");
+        when(freeIpaService.findByStack(stack)).thenReturn(freeIpa);
+        InstanceTemplate template = mock(InstanceTemplate.class);
+        when(template.getGroupName()).thenReturn(GROUP_NAME);
+        when(template.getPrivateId()).thenReturn(INSTANCE_PRIVATE_ID_3);
+        List<CloudInstance> cloudInstances = List.of(new CloudInstance(INSTANCE_ID_3, template, null, "subnet-1", "az1"));
+        DetailedEnvironmentResponse environmentResponse = new DetailedEnvironmentResponse();
+        when(cachedEnvironmentClientService.getByCrn(ENVIRONMENT_ID)).thenReturn(environmentResponse);
+        Map<String, String> subnetAzMap = Map.of();
+        when(multiAzCalculatorService.prepareSubnetAzMap(environmentResponse)).thenReturn(subnetAzMap);
+        Map<String, Integer> subentUsage = Map.of();
+        when(multiAzCalculatorService.calculateCurrentSubnetUsage(subnetAzMap, instanceGroup)).thenReturn(subentUsage);
+
+        Stack stack1 = underTest.saveInstanceAndGetUpdatedStack(stack, cloudInstances);
```

Comment:
Nit: I see this is copied from above, but a more descriptive name like `updatedStack` would sound better. :smile: